src/apps/testapps/testCompactCells.c

```diff
@@ -70,6 +70,26 @@ SUITE(compactCells) {
         free(sunnyvaleExpanded);
     }
 
+    TEST(res0children) {
+        H3Index parent;
+        setH3Index(&parent, 0, 0, 0);
+
+        int64_t arrSize;
+        t_assertSuccess(H3_EXPORT(cellToChildrenSize)(parent, 1, &arrSize));
+
+        H3Index *children = calloc(arrSize, sizeof(H3Index));
+        t_assertSuccess(H3_EXPORT(cellToChildren)(parent, 1, children));
+
+        t_assert(arrSize > 0, "got expected number of children");
+        for (int i = 0; i < arrSize; i++) {
+            t_assert(children[i] != 0, "got expected child");
+        }
+
+        H3Index *compressed = calloc(1, sizeof(H3Index));
```

Comment:
Doesn't `compressed` need to be of size arrSize per the function contract? https://h3geo.org/docs/api/hierarchy/#compactcells

Also, please add corresponding `free` calls for `calloc`.

Reply:
good call, I wrote this fast. it should be yes, and I will change that but that doesn't affect the integrity of the test as there *should* only be 1 cell returned.

Reply:
updated